File: DataBases/sem6/lab1/cmd/p1/indexes.go
package main

import (
	"db_lab1/internal/db"
	"flag"
	"log"
)

func main() {
	log.SetFlags(log.LstdFlags | log.Lshortfile)

	mode := flag.String("mode", "benchmark-index", "режим: create-indexes, drop-indexes, benchmark-index, benchmark-noindex")
	flag.Parse()

	store, err := db.NewStore()
	if err != nil {
		log.Fatalf("Ошибка подключения к БД: %v", err)
	}
	defer store.Close()

	switch *mode {

	case "create-indexes":
		if err := store.CreateIndexes(); err != nil {
			log.Fatalf("Ошибка при создании индексов: %v", err)
		}

	case "drop-indexes":
		if err := store.DropIndexes(); err != nil {
			log.Fatalf("Ошибка при удалении индексов: %v", err)
		}

	case "benchmark-index":
		if err := store.Benchmark("с индексами", true); err != nil {
			log.Fatalf("Ошибка бенчмарка с индексами: %v", err)
		}

	case "benchmark-noindex":
		if err := store.Benchmark("без индексов", false); err != nil {
			log.Fatalf("Ошибка бенчмарка без индексов: %v", err)
		}

	default:
		log.Fatalf("Неизвестный режим: %s", *mode)
	}
}

File: DataBases/sem6/lab1/cmd/p1/indexes.py
import os
import sys
import logging
import argparse

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..')))

from internal.db.store import Store

def main():
    logging.basicConfig(level=logging.INFO)

    parser = argparse.ArgumentParser()
    parser.add_argument("--mode", default="Benchmark-Index", help="режимы: Create-Noindexes, Drop-Indexes, Benchmark-Index, Benchmark-Noindex")
    args = parser.parse_args()

    store = Store()
    try:
        if args.mode == "Create-Indexes":
            print("=== Create-Indexes ===")
            store.create_indexes()
        elif args.mode == "Drop-Indexes":
            print("=== Drop-Indexes ===")
            store.drop_indexes()
        elif args.mode == "Benchmark-Index":
            print("=== Benchmark-Index ===")
            store.benchmark("с индексами", use_index=True)
        elif args.mode == "Benchmark-Noindex":
            print("=== Benchmark-Noindex ===")
            store.benchmark("без индексов", use_index=False)
        else:
            raise ValueError(f"Неизвестный режим: {args.mode}")
    finally:
        store.close()

if __name__ == "__main__":
    main()